### PR TITLE
DAOS-10030 doc: fix a few more doxygen rendering issues

### PR DESCRIPTION
**Pull request by kccain**
Replace "# " with "number of".
Add a link to existing structure member rather than \a emphasis.

Skip-test: true

Signed-off-by: Kenneth Cain <kenneth.c.cain@intel.com>

### CODE REVIEW

File: src/include/daos_pool.h [hunk 1]
```diff
@@ -105,15 +105,15 @@ struct daos_rebuild_status {
...
-	/** # total to-be-rebuilt objects, it's non-zero and increase when
-	 * rebuilding in progress, when rs_done is 1 it will not change anymore
+	/** total number of objects to be rebuilt. Non-zero and increases when
+	 * rebuilding in progress. When rs_done is 1 it will not change anymore
 	 * and should equal to rs_obj_nr. With both rs_toberb_obj_nr and
 	 * rs_obj_nr the user can know the progress of the rebuilding.
 	 */
 	uint64_t		rs_toberb_obj_nr;
-	/** # rebuilt objects, it's non-zero only if rs_done is 1 */
+	/** number of rebuilt object. Non-zero only if rs_done is 1 */
```
Comment (Michael-Hennecke) [inline src/include/daos_pool.h:114]: objects ?

File: src/include/daos_pool.h [hunk 2]
```diff
@@ -105,15 +105,15 @@ struct daos_rebuild_status {
...
 	/* Failure on which rank */
 	int32_t			rs_fail_rank;
-	/** # total to-be-rebuilt objects, it's non-zero and increase when
-	 * rebuilding in progress, when rs_done is 1 it will not change anymore
+	/** total number of objects to be rebuilt. Non-zero and increases when
+	 * rebuilding in progress. When rs_done is 1 it will not change anymore
 	 * and should equal to rs_obj_nr. With both rs_toberb_obj_nr and
 	 * rs_obj_nr the user can know the progress of the rebuilding.
 	 */
 	uint64_t		rs_toberb_obj_nr;
```
[the remark this answers is not in the document]
Comment (kccain) [reply src/include/daos_pool.h:130]: Not sure why, but something that jumps out as a possibility also could be rs_objs_to_rebuild_nr or rs_obj_to_rebuild_nr.

**Comment by kccain** [comment]
In most recent commit (that produced build 2 testing https://build.hpdd.intel.com/blue/organizations/jenkins/daos-stack%2Fdaos/detail/PR-8404/2/) I forgot to use the Skip-test: true commit pragma. So I have manually stopped that build.

PR should be ready for a final review.

**Comment by kccain** [comment]
Just a reminder this PR needs review.

**Review by ashleypittman — APPROVED** [review]
Looks reasonable to me.

I think we should investigate to see if there's a way to view the rendered files as part of the PR build data as just because the markup is valid it doesn't mean it's correct.

**Review by daltonbohning — APPROVED** [review]

**Comment by kccain** [comment]
> Looks reasonable to me.
> 
> I think we should investigate to see if there's a way to view the rendered files as part of the PR build data as just because the markup is valid it doesn't mean it's correct.

For the immediate purposes I did confirm just now using a manual doxygen run into my local tree docs/doxygen folder that the rendering worked as intended for these changes. I'm not immediately familiar with GitHub Actions and how one can (or cannot) store and inspect the artifacts -- or how one might choose to save the rendered documentation somewhere else e.g., build..hpdd.intel.com.

**Comment by ashleypittman** [comment]
> > Looks reasonable to me.
> > I think we should investigate to see if there's a way to view the rendered files as part of the PR build data as just because the markup is valid it doesn't mean it's correct.
> 
> For the immediate purposes I did confirm just now using a manual doxygen run into my local tree docs/doxygen folder that the rendering worked as intended for these changes. I'm not immediately familiar with GitHub Actions and how one can (or cannot) store and inspect the artifacts -- or how one might choose to save the rendered documentation somewhere else e.g., build..hpdd.intel.com.

No, it's on my list to look at when I'm not busy with other things but it might be a while until I get round to it.

**Review by johannlombardi — APPROVED** [review]

**Pull request merged**